### PR TITLE
cmd/cueckoo: changes to support new Dispatch-Trailer CI flows

### PR DESCRIPTION
**Pull request by myitcv**
(see commit message)

### CODE REVIEW

File: cmd/cueckoo/cmd/cltrigger.go [hunk 1]
```diff
@@ -262,24 +262,20 @@ func (c *cltrigger) triggerBuild(rev revision) error {
...
-		// as tildes are not allowed in
-		// Use the change ID alone, without any tildes.
-		ChangeID: in.ChangeID,
-
-		Ref:    revision.Ref,
-		Commit: commit,
-		Branch: in.Branch,
+	return c.builder(repositoryDispatchPayload{
+		CL:           in.ChangeNumber,
+		Patchset:     in.Revisions[in.CurrentRevision].PatchSetNumber,
```
Comment (mvdan) [inline cmd/cueckoo/cmd/cltrigger.go:267]: I assume we want `Patchset` and `Ref` to be consistent, so then this should be `revision.PatchSetNumber`.

File: cmd/cueckoo/cmd/runtrybot.go [hunk 2]
```diff
@@ -68,16 +68,20 @@ func runtrybotDef(cmd *Command, args []string) error {
 	if err != nil {
 		return err
 	}
-	r := newCLTrigger(cmd, cfg, func(payload clTriggerPayload) error {
-		p, err := buildTryBotPayload(payload)
+	r := newCLTrigger(cmd, cfg, func(payload repositoryDispatchPayload) error {
+		trybotPayload := payload
+		trybotPayload.Type = string(eventTypeTrybot)
```
Comment (mvdan) [inline cmd/cueckoo/cmd/runtrybot.go:73]: make the Type field use the same named type for consistency?

**Review by mvdan — APPROVED** [review]

File: cmd/cueckoo/cmd/cltrigger.go (hunk 1 above)
Comment (myitcv) [reply cmd/cueckoo/cmd/cltrigger.go:267]: Wow, not sure how I missed this above. Thanks. Done

File: cmd/cueckoo/cmd/runtrybot.go (hunk 2 above)
Comment (myitcv) [reply cmd/cueckoo/cmd/runtrybot.go:73]: I delibeately did not do that in order that the `repositoryDispatchPayload` more closely matches the CUE type which is the source of truth. There's more consolidation/refactoring we could do there, but this is sufficient for now IMO.

**Comment by myitcv** [comment]
DO NOT SUBMIT until https://review.gerrithub.io/c/cue-lang/cue/+/551352 has landed and we have started the "release".

**Pull request merged**